Automattic/jetpack
# Protect: Refresh plan data after returning from checkout

Fixes an issue where the free UI would still be displayed on the initial page load after purchasing a paid plan, by including a `checkPlan` URL parameter in the checkout redirect so the application knows it needs to refresh the plan data.

#### Changes proposed in this Pull Request:

* Adds `@wordpress/url` dependency
* Adds `checkPlan` to the redirect URL for upgrading Protect
* Checks for `checkPlan` query param in the Admin page and refreshes plan data when present.

#### Other information:


#### Jetpack product discussion



1201069996155224-as-1203203146712456/f

#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






* Spin up a JN site
* Go to the Protect admin screen and click "Get Jetpack Protect", follow checkout flow
* When you return to the app, the URL should include a `checkPlan` query parameter, and the paid UI should be shown (i.e. can see Scan button)
